src/ci/tested_versions_utils.py

```diff
@@ -151,7 +152,9 @@ def parse_version(version: str) -> Union[SemanticVersion, NonSemanticVersion]:
 
 @dataclass(frozen=True)
 class TestedVersions:
-    versions: List[Union[SemanticVersion, NonSemanticVersion]]
+    versions: List[Union[SemanticVersion, NonSemanticVersion]] = attrib(
```

Comment:
Cool! 💪

Reply:
Turns out `attr.s` has its own flavor of `frozen` that we can use instead of the `@dataclass`. And since we don't need sorting or equality checks or anything else for `TestedVersions`, we might as well jump over for this case.